src/features/accessibility/components/AccessibilityFiltersModal.native.test.tsx

```diff
@@ -0,0 +1,144 @@
+}
+
+jest.mock('features/accessibility/context/AccessibilityFiltersWrapper')
+const mockUseAccessibilityFiltersContext = useAccessibilityFiltersContext as jest.Mock
+
+describe('<AccessibilityFiltersModal />', () => {
+  beforeEach(() => {
+    mockUseAccessibilityFiltersContext.mockReturnValue(defaultValuesAccessibilityContext)
+  })
+
+  it('should render modal correctly', async () => {
+    renderAccessibilityFiltersModal()
+    await screen.findByText(
+      'Filtrer par l’accessibilité des lieux en fonction d’un ou plusieurs handicaps'
+    )
+
+    expect(screen).toMatchSnapshot()
+  })
+
+  it('should call context when modal is opened', async () => {
+    renderAccessibilityFiltersModal()
+    await screen.findByText(
+      'Filtrer par l’accessibilité des lieux en fonction d’un ou plusieurs handicaps'
+    )
+
+    expect(mockUseAccessibilityFiltersContext).toHaveBeenCalledWith()
+  })
+
+  it('should call context when search button is pressed', async () => {
+    const context = mockUseAccessibilityFiltersContext.mockReturnValue(
+      defaultValuesAccessibilityContext
+    )
+    const { disabilities, setDisabilities } = context()
+    renderAccessibilityFiltersModal()
+
+    const audioCheckbox = await screen.findByRole('checkbox', { name: 'Handicap auditif' })
+    fireEvent.press(audioCheckbox)
+
+    const searchButton = await screen.findByTestId('Rechercher')
+    fireEvent.press(searchButton)
+
+    expect(setDisabilities).toHaveBeenCalledWith({
+      ...disabilities,
+      isAudioDisabilityCompliant: true,
+    })
+  })
+
+  it('should not call context when search button is not pressed', async () => {
+    const context = mockUseAccessibilityFiltersContext.mockReturnValue(
+      defaultValuesAccessibilityContext
+    )
+    const { disabilities, setDisabilities } = context()
+    renderAccessibilityFiltersModal()
+
+    const audioCheckbox = await screen.findByRole('checkbox', { name: 'Handicap visuel' })
+    fireEvent.press(audioCheckbox)
+
+    const closeButton = await screen.findByTestId('icon-close')
+    fireEvent.press(closeButton)
+
+    expect(setDisabilities).not.toHaveBeenCalledWith({
+      ...disabilities,
+      isVisualDisabilityCompliant: true,
+    })
+  })
+
+  it('should reset filter when reset button and search button are pressed', async () => {
+    const context = mockUseAccessibilityFiltersContext.mockReturnValue({
+      disabilities: { ...mockDisabilitiesTrue },
+      setDisabilities: jest.fn(() => ({})),
+    })
+
+    const { setDisabilities } = context()
+    renderAccessibilityFiltersModal()
+
+    const resetButton = await screen.findByTestId('Réinitialiser')
+    fireEvent.press(resetButton)
+
+    const searchButton = await screen.findByTestId('Rechercher')
+    fireEvent.press(searchButton)
+
+    expect(setDisabilities).toHaveBeenCalledWith(mockDisabilitiesFalse)
+  })
+
+  it('should not reset filter when reset button is pressed but search button is not', async () => {
```

Comment:
selon moi le test devrait plutôt être should keep previous filter when close button is pressed:
- ouvrir la modal
- modifier les cases cochées
- appuyer sur la croix
- reouvrir la modal
- vérifier que on a bien les cases cochées du début